<!--
Runs the RTABMap SLAM node.

Requires:

    sudo apt-get install ros-kinetic-rtabmap-ros

http://wiki.ros.org/rtabmap_ros/Tutorials/SetupOnYourRobot
-->
<launch>
  <group ns="rtabmap">
    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
          <param name="frame_id" type="string" value="base_link"/>

          <param name="subscribe_depth" type="bool" value="true"/>
          <param name="subscribe_laserScan" type="bool" value="true"/>

          <remap from="odom" to="/base_controller/odom"/>
          <remap from="scan" to="/base_scan"/>

          <remap from="rgb/image" to="/camera/rgb/image_rect_color"/>
          <remap from="depth/image" to="/camera/depth_registered/image_raw"/>
          <remap from="rgb/camera_info" to="/camera/rgb/camera_info"/>

          <param name="queue_size" type="int" value="10"/>

          <!-- RTAB-Map's parameters -->
          <param name="RGBD/PoseScanMatching" type="string" value="true"/>
          <param name="RGBD/LocalLoopDetectionSpace" type="string" value="true"/>
          <param name="LccIcp/Type" type="string" value="2"/>     <!-- 2=ICP 2D -->
          <param name="LccBow/MinInliers" type="string" value="5"/>
          <param name="LccBow/InlierDistance" type="string" value="0.1"/>
          <param name="RGBD/AngularUpdate" type="string" value="0.01"/>
          <param name="RGBD/LinearUpdate" type="string" value="0.01"/>
          <param name="Rtabmap/TimeThr" type="string" value="700"/>
          <param name="Mem/RehearsalSimilarity" type="string" value="0.45"/>
          <param name="RGBD/OptimizeFromGraphEnd" type="string" value="true"/>
    </node>
  </group>
</launch>
